### ISSUE TITLE
Linq with FetchLazyProperties() resets lazy property changes

### ISSUE THREAD

**Issue by AlexanderShutov** [nhibernate/nhibernate-core#3330]
In my project, I got a situation with bad behavior of linq with FetchLazyProperties().
FetchLazyProperties() resets lazy property changes.

I have written a test for this problem:
https://github.com/nhibernate/nhibernate-core/pull/3329

I think this is not correct behavior

**Comment by bahusoid** [comment]
It's a 5.3 regression in hql. 

hql versions of test works with 5.2.7 and fails with 5.3+

```C#
//var allPersons = s.Query<Person>().FetchLazyProperties().ToList();
var allPersons = s.CreateQuery("from Person fetch all properties").List<Person>();
```

**Comment by bahusoid** [comment]
As a workaround use explicit fetch for lazy properties:
```C#
var allPersons = s.Query<Person>().Fetch(p => p.Image).ToList();
```

**Comment by fredericDelaporte** [comment]
So we need a 5.3.18, milestone added. 5.4.3 is at 100% and could be released, but I am considering waiting for this fix and the release of 5.3.18, to avoid having a 5.4.4 to do little time afterward.

**Comment by bahusoid** [comment]
Fixed by #3343
